sdk/digitaltwins/azure-digitaltwins-core/src/main/java/com/azure/digitaltwins/core/IDigitalTwin.java

```diff
@@ -0,0 +1,54 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.digitaltwins.core;
+
+import com.azure.digitaltwins.core.models.DigitalTwinMetadata;
+import com.fasterxml.jackson.annotation.JsonProperty;
+
+/**
+ * Interface that defines the Json properties that are present in every digital twin. Implementations of this class
+ * must add Json properties for each of the getters defined in this interface. See each getter for more
+ * details on each Json property.
+ */
+public interface IDigitalTwin {
+    public static String DIGITAL_TWIN_ID_JSON_PROPERTY_NAME = "$dtId";
+    public static String DIGITAL_TWIN_ETAG_JSON_PROPERTY_NAME = "$etag";
+    public static String DIGITAL_TWIN__METADATA_JSON_PROPERTY_NAME = "$metadata";
```

Comment:
@JonathanGiles For these interfaces, I want to expose the json property names in the interface itself so that people have an easy hook to use when they write their own DigitalTwin implementations, but I wanted to check if this was bad practice first.

Reply:
The same values can be documented just in the headers of the getters below, but this feels a bit more useable to me